app/components/Form/InlineNumber/InlineNumber.tsx

```diff
@@ -177,18 +178,25 @@ export default function InlineNumber({
 
     const handleClick: MouseEventHandler<HTMLInputElement> = useCallback(
         (e) => {
-            if (clearOnClick && e.currentTarget.value === formattedFallback) {
+            if (clearOnFocus && e.currentTarget.value === formattedFallback) {
                 setInnerValue('');
                 e.currentTarget.style.width = '6px';
             }
         },
-        [formattedFallback, clearOnClick]
+        [formattedFallback, clearOnFocus]
     );
 
-    const handleFocus = useCallback(() => {
-        setIsFocused(true);
-        onFocus();
-    }, [onFocus]);
+    const handleFocus: FocusEventHandler<HTMLInputElement> = useCallback(
+        (e) => {
+            setIsFocused(true);
+            if (clearOnFocus && e.currentTarget.value === formattedFallback) {
+                setInnerValue('');
+                e.currentTarget.style.width = '6px';
+            }
```

Comment:
Maybe we should make extract this into a function, now that it's used twice?

Reply:
Although... can't we just remove the onclick handler now? The focus handler should also be executed when the input is clicked, as it also triggers focus?